labs/projecteuler/projecteuler.md

```diff
@@ -0,0 +1,132 @@
+
+ONE IMPORTANT NOTE: we have reworded the questions to make them more Cryptol-friendly.  The Project Euler questions often ask you to find all the numbers which satisfy some property and then combine them in some way (e.g. take their collective sum).  For most of the problems, it is hard enough to get Cryptol to find an answer, let alone all possible answers.  For these exercises, it is enough to write code that will find an answer, not all of them.
+
+### [Problem 9](https://projecteuler.net/problem=9)
+
+> A Pythagorean triplet is a set of three natural numbers, a < b < c, for which a^2 + b^2 = c^2.
+> 
+> For example, 3^2 + 4^2 = 9 + 16 = 25 = 5^2.  There exists exactly one Pythagorean triplet for which a + b + c = 1000.  Find this triple.
+
+### [Problem 34](https://projecteuler.net/problem=34)
+
+> 145 is a curious number, as 1! + 4! + 5! = 1 + 24 + 120 = 145.
+>
+> Find all numbers which are equal to the sum of the factorial of their digits.
+> 
+> Note: as 1! = 1 and 2! = 2 are not sums they are not included.
+>
+> (Aside: these numbers are called [factorions](https://en.wikipedia.org/wiki/Factorion).  It is unclear why Project Euler does not include 1 and 2).
+>
+> Hints: the factorial function is usually defined recursively, but that tends to make SAT solving difficult.  Since you only need to calculate the factorial of the numbers 0-9, make your function just do a case by case calculation.  To get the digital representation of the number, create a function which takes in a number and a list of numbers and returns true exactly when the list is the base 10 representation.  Finally, it can be shown that the most number of di\
+gits a factorion can have is 6.
+
+### [Problem 36](https://projecteuler.net/problem=36)
+
+> The decimal number, 585 = 1001001001_(2) (binary), is palindromic in both bases.
+>
+> Find at least three numbers which are palindromic in base 10 and base 2.
+>
+> (Please note that the palindromic number, in either base, may not include leading zeros.)
+
+### [Problem 43](https://projecteuler.net/problem=43)
+
+> The number, 1406357289, is a 0 to 9 pandigital number because it is made up of each of the digits 0 to 9 in some order, but it also has a rather interesting sub-string divisibility property.
+>
+> Let d_(1) be the 1^(st) digit, d_(2) be the 2^(nd) digit, and so on. In this way, we note the following:
+>
+>    * d_(2)d_(3)d_(4)=406 is divisible by 2
+>    * d_(3)d_(4)d_(5)=063 is divisible by 3
+>    * d_(4)d_(5)d_(6)=635 is divisible by 5
+>    * d_(5)d_(6)d_(7)=357 is divisible by 7
+>    * d_(6)d_(7)d_(8)=572 is divisible by 11
+>    * d_(7)d_(8)d_(9)=728 is divisible by 13
+>    * d_(8)d_(9)d_(10)=289 is divisible by 17
+>
```

Comment:
Hmmm... it seems that my markdown browser plugin doesn't interpret the sub-/superscript correctly.  Use the online renderer as suggested by Sean, e.g.: 
> ![](https://render.githubusercontent.com/render/math?math=R(x)%20=%20M(x)%20\cdot%20x^n%20\\bmod\%20G(x))

Reply:
I'm not sure the renderer gives that much value for how difficult it is to use.